test/PoolManager.clear.t.sol

```diff
@@ -0,0 +1,191 @@
+        actions[4] = Actions.CLEAR;
+        params[4] = abi.encode(currency0, amount + 1);
+
+        vm.expectRevert(IPoolManager.MustClearExactBalance.selector);
+        actionsRouter.executeActions(actions, params);
+    }
+
+    function test_clear_reverts_positiveDelta_inputLessThanDelta() external {
+        uint256 amount = 1e18;
+        Actions[] memory actions = new Actions[](5);
+        bytes[] memory params = new bytes[](5);
+
+        // Apply a positive delta.
+        actions[0] = Actions.SYNC;
+        params[0] = abi.encode(currency0);
+
+        actions[1] = Actions.TRANSFER_FROM;
+        params[1] = abi.encode(currency0, address(this), address(manager), amount);
+
+        actions[2] = Actions.SETTLE;
+        params[2] = abi.encode(currency0);
+
+        actions[3] = Actions.ASSERT_DELTA_EQUALS;
+        params[3] = abi.encode(currency0, address(actionsRouter), amount);
+
+        // Clear for 1 less than the delta.
+        actions[4] = Actions.CLEAR;
+        params[4] = abi.encode(currency0, amount - 1);
+
+        vm.expectRevert(IPoolManager.MustClearExactBalance.selector);
+        actionsRouter.executeActions(actions, params);
+    }
+
+    function test_clear_reverts_positiveDelta_inputZero() external {
+        uint256 amount = 1e18;
+        Actions[] memory actions = new Actions[](5);
+        bytes[] memory params = new bytes[](5);
+
+        // Apply a positive delta.
+        actions[0] = Actions.SYNC;
+        params[0] = abi.encode(currency0);
+
+        actions[1] = Actions.TRANSFER_FROM;
+        params[1] = abi.encode(currency0, address(this), address(manager), amount);
+
+        actions[2] = Actions.SETTLE;
+        params[2] = abi.encode(currency0);
+
+        actions[3] = Actions.ASSERT_DELTA_EQUALS;
+        params[3] = abi.encode(currency0, address(actionsRouter), amount);
+
+        // Clear with 0.
+        actions[4] = Actions.CLEAR;
+        params[4] = abi.encode(currency0, 0);
+
+        vm.expectRevert(IPoolManager.MustClearExactBalance.selector);
+        actionsRouter.executeActions(actions, params);
+    }
+
+    function test_clear_zeroDelta_inputZero_isUnchanged() external {
+        Actions[] memory actions = new Actions[](5);
+        bytes[] memory params = new bytes[](5);
+
+        actions[0] = Actions.ASSERT_NONZERO_DELTA_COUNT_EQUALS;
+        params[0] = abi.encode(0);
+
+        actions[1] = Actions.ASSERT_DELTA_EQUALS;
+        params[1] = abi.encode(currency0, address(actionsRouter), 0);
+
+        // Clear with 0.
+        actions[2] = Actions.CLEAR;
+        params[2] = abi.encode(currency0, 0);
+
+        actions[3] = Actions.ASSERT_DELTA_EQUALS;
+        params[3] = abi.encode(currency0, address(actionsRouter), 0);
+
+        actions[4] = Actions.ASSERT_NONZERO_DELTA_COUNT_EQUALS;
+        params[4] = abi.encode(0);
+
+        actionsRouter.executeActions(actions, params);
+    }
+
+    function test_clear_reverts_zeroDelta_inputNonZero() external {
```

Comment:
Can I get some eyes on the behavior for this case? If the open delta is 0 and you call clear with anything nonzero it reverts with `MustClearExactBalance`.